Classes/Emogrifier.php

```diff
@@ -681,8 +692,22 @@ private function getUnifiedHtml()
         } else {
             $bodyWithoutUnprocessableTags = $this->html;
         }
+        
+        if ($this->shouldFixMissingCharsetDeclaration) {
+            if (!stristr($bodyWithoutUnprocessableTags, 'Content-Type')) {
+
+                // We are trying to insert the meta tag to the right spot in the DOM as if we just prepend it to the HTML we will lose attributes set to the HTML tag.
+                if (stristr($bodyWithoutUnprocessableTags, '<head')) {
+                    $bodyWithoutUnprocessableTags = preg_replace('/<head(.*?)>/i', '<head$1><meta http-equiv="Content-Type" content="text/html; charset=utf-8">', $bodyWithoutUnprocessableTags);
+                } elseif (stristr($bodyWithoutUnprocessableTags, '<html')) {
+                    $bodyWithoutUnprocessableTags = preg_replace('/<html(.*?)>/i', '<html$1><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"></head>', $bodyWithoutUnprocessableTags);
```

Comment:
We have duplicate code here (at least the meta tag). Please move it to a local variable.